File: ABC/ABC301-400/ABC365/D/D.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int N;
	string S;
	cin >> N >> S;
	int r = 0, s = 0, p = 0;
	for (char c: S) {
		tie(r, s, p) = make_tuple(max(s, p), max(p, r), max(r, s));
		if (c == 'R') {
			s = 0;
			p++;
		}
		if (c == 'S') {
			p = 0;
			r++;
		}
		if (c == 'P') {
			r = 0;
			s++;
		}
	}
	cout << ranges::max({r, s, p}) << '\n';
}

File: ABC/ABC301-400/ABC365/D/D.py
N = int(input())
S = input()

r, s, p = 0, 0, 0
for c in S:
	r, s, p = max(s, p), max(p, r), max(r, s)
	if c == 'R':
		s = 0
		p += 1
	if c == 'S':
		p = 0
		r += 1
	if c == 'P':
		r = 0
		s += 1
print(max(r, s, p))